magda-registry-aspects/provenance.schema.json

```diff
@@ -13,23 +13,45 @@
             "type": "string"
         },
         "derivedFrom": {
-            "description": "What other record(s), if any, was this data derived from?",
-            "type": "object",
-            "properties": {
-                "id": {
-                    "description": "The id(s) of the record(s)",
-                    "type": "array",
-                    "items": {
-                        "description": "The id of the record",
+            "description": "What other dataset(s), if any, was this data derived from?",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "description": "The dataset that this data was derived from - either a link to another record in Magda or a name if the dataset isn't in Magda",
+                "properties": {
+                    "id": {
+                        "type": "array",
```

Comment:
Why `array` type? Seems only a string in database:
```json
{
  "derivedFrom": [
    {
      "id": "ds-qld-03a60187-37a2-48e8-9cfa-9c7f89ddd7ef"
    },
    {
      "id": "magda-ds-062eb059-00b2-4be7-96ce-c0ae1cd6a1f2"
    },
    {
      "name": "Use name: \"test use the name\""
    }
  ]
}
```

Reply:
The array type is because it's not entirely clear how non-array links should be dereferenced (especially with regards to authn). It doesn't get set as an array because I messed it up 😂 .